### ISSUE TITLE
open_mfdataset not decoding times correctly

### ISSUE THREAD

**Issue by pochedls** [XCDAT/xcdat#158]
### What versions of software are you using?

-   Package Version: main branch (latest)

### What are the steps to reproduce this issue?
```
import xarray as xr
import xcdat

dpath = "/p/css03/cmip5_css02/data/cmip5/output1/CMCC/CMCC-CESM/historical/mon/atmos/Amon/r1i1p1/tas/1/*nc"

# xcdat
ds = xcdat.open_mfdataset(dpath, "tas")
print('xcdat inferred tas shape: ', ds.tas.shape)
ds.close()

# xarray
ds = xr.open_mfdataset(dpath)
print('xarray inferred tas shape: ', ds.tas.shape)
ds.close()

# xarray with decode_times=False
ds = xr.open_mfdataset(dpath, decode_times=False)
print('xarray (no time decoding) inferred tas shape: ', ds.tas.shape)
ds.close()
```
### What happens? Any logs, error output, etc?

The code excerpt above shows that xcdat yields a `tas` time series that has only 12 time steps, but xarray has 1872. We can reproduce the xcdat result by adding `decode_times=False` (which is included in the `xcdat.open_mfdataset()` logic [here](https://github.com/XCDAT/xcdat/blob/main/xcdat/dataset.py#L172).

> xcdat inferred tas shape:  (12, 48, 96)
> xarray inferred tas shape:  (1872, 48, 96)
> xarray (no time decoding) inferred tas shape:  (12, 48, 96)

### What were you expecting to happen?

`xcdat.open_mfdataset()` should be able to load and decode the entire time series (1872 time steps). 

### Any other comments?

I think (this is a guess) this may arise because the time units in each netCDF in this directory have different units (but perhaps the same time values). We may need to add a function that checks for this and reverts to `decode_times=False`. I imagine this could get complicated in some situations, but I think starting by turning this off in some circumstances would likely be the most ideal behavior.

**Comment by pochedls** [comment]
I was able to reproduce this with a different model (that also had different time units in each file). 

I wonder if we could modify `open_mfdataset` so that it:

1. Opens the dataset with as little overhead as possible (e.g., without decoding time and perhaps opening only one file)
2. Checks the time units (if time is in the file)
3. Closes that test dataset (if need be)
3. If the units are `years` or `months since XXXX-XX-XX`, then open the dataset with `decode_times=False` (and then call `decode_time_units(ds)`), else open the dataset with `decode_times=True`

**Comment by pochedls** [comment]
@tomvothecoder - I made an initial attempt at #158. One decision I made was to only open one file in multi-file datasets (there is a 30x difference in using `open_dataset` versus `open_mfdataset` in my example). I think this is okay for our purposes, though it could potentially be extended (e.g., with an optional flag `check_all_files=False`). There are a couple notes that would benefit from your comments.
